shared/src/commonMain/kotlin/com/devil/phoenixproject/presentation/screen/RoutineOverviewScreen.kt

```
@@ -114,6 +123,28 @@ fun RoutineOverviewScreen(navController: NavController, viewModel: MainViewModel
 
+    // Issue #190: Auto-start routine — skip overview and enter SetReady for exercise 0
+    // Uses a one-shot flag to prevent re-triggering on recomposition or back-navigation.
+    // Guard conditions: pref enabled, routine non-empty, BLE connected, no resumable progress.
+    var autoStartFired by remember { mutableStateOf(false) }
+    LaunchedEffect(routine.id, userPreferences.autoStartRoutine) {
+        if (
+            !autoStartFired &&
+            userPreferences.autoStartRoutine &&
+            routine.exercises.isNotEmpty() &&
+            connectionState is ConnectionState.Connected &&
+            !viewModel.hasResumableProgress(routine.id)
```

detekt (1.23.8)

[warning] This condition is too complex (5). Defined complexity threshold for conditions is set to '4'

(detekt.complexity.ComplexCondition)